### ISSUE TITLE
[array-callback-return] an async function should be considered a proper return. 

### ISSUE THREAD

**Issue by tikotzky** [eslint/eslint#5701]
**What version of ESLint are you using?**
v2.5.1

**What parser (default, Babel-ESLint, etc.) are you using?**
babel-eslint

**Please show your full configuration:**

``` json
{
  "parser": "babel-eslint",
  "rules": {
    "array-callback-return": 2
  }
}
```

**What did you do? Please include the actual source code causing the issue.**

``` js
var numbers = [1,2,3];
var promises = numbers.map(async function(num) {
  await somethingAsync(num)
});
Promise.all(promises).then(function() {
  console.log('All async done')
});
```

**What did you expect to happen?**
Expected to get no errors from eslint since `async` functions implicitly `return` a promise which can be later used by `Promise.all`

**What actually happened? Please include the actual, raw output from ESLint.**

```
  2:28  error  Expected to return a value in this function  array-callback-return
```

**Comment by ilyavolodin** [comment]
ESLint doesn't support rules for experimental features. You should open an issue for [eslint-plugin-babel](https://github.com/babel/eslint-plugin-babel). Closing as this is outside of the scope of the core.

**Comment by vkurchatkin** [comment]
@ilyavolodin is async still considered experimental? I can see http://eslint.org/docs/rules/no-return-await

**Comment by platinumazure** [comment]
@vkurchatkin No, async/await now exists in the ES2017 specification, so we should reopen this.

**Comment by vkurchatkin** [comment]
@platinumazure It seems to be fixed already: https://github.com/eslint/eslint/commit/f71937af3046584fdec8cc928ed222e1f0b8f11d, sorry about that

**Comment by not-an-aardvark** [comment]
Yes, this is already fixed as of v3.6.0. Closing this issue.

**Comment by platinumazure** [comment]
My apologies, I should have investigated that. Thanks.
